### ISSUE TITLE
auto transform does not work for negative bounds, when bounds are too large

### ISSUE THREAD

**Issue by flo-schu** [fau-advanced-separations/CADET-Process#37]
```python
from CADETProcess.optimization import OptimizationProblem

optimization_problem = OptimizationProblem('simple')

optimization_problem.add_variable(
    f'var_negpos', 
    lb=-1000, ub=1000,
    transform="auto"
)

optimization_problem.add_objective(lambda x: x**2, "test")

optimization_problem.evaluate_objectives([0.5], untransform=True)

```

raises an error when the bounds get too large. The problem does not occur when the bounds are e.g. [-100, 100]

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/flo-schu/repositories/CADET-Process/CADETProcess/transform.py", line 201, in untransform
    x = self._untransform(x)
  File "/home/flo-schu/repositories/CADET-Process/CADETProcess/transform.py", line 526, in _untransform
    return self.log.untransform(x)
  File "/home/flo-schu/repositories/CADET-Process/CADETProcess/transform.py", line 201, in untransform
    x = self._untransform(x)
  File "/home/flo-schu/repositories/CADET-Process/CADETProcess/transform.py", line 395, in _untransform
    np.exp(x * np.log(self._ub - self.lb_input + 1)) \
AttributeError: 'NormLogTransform' object has no attribute '_ub'
```

**Comment by schmoelder** [comment]
Fixed in 851e0ce0001f6dd58b2b2f0cc9e7800ec417837e.

It's important to note that a log transform is generally impossible for values with any bounds < 0. To still allow some kind of log transform, the parameter space is shifted by the amount of the lower value and then a regular log transform between lb' and ub' is performed. Hope that makes sense.

**Comment by flo-schu** [comment]
for transformations from unconstrained scale to [0,1] typically logit transform is used. Although this does not apply to bounded problems

https://en.wikipedia.org/wiki/Logit
